groups/nts/ntsu/ntsu_timestamputil.cpp

```diff
@@ -28,9 +32,66 @@
 #endif
 // clang-format on
 
+#if defined(BSLS_PLATFORM_OS_LINUX)
+
+#define NTSU_TIMESTAMP_UTIL_LINUX_VERSION(major, minor, patch) \
+    static_cast<bsl::size_t>( \
+        KERNEL_VERSION( \
+            static_cast<bsl::size_t>(versionMajor), \
+            static_cast<bsl::size_t>(versionMinor), \
+            static_cast<bsl::size_t>(versionPatch)))
+
+#define NTSU_TIMESTAMP_UTIL_LINUX_VERSION_GE(version, major, minor, patch) \
+    (static_cast<bsl::size_t>(version)) >= \
+    (NTSU_TIMESTAMP_UTIL_LINUX_VERSION(major, minor, patch))
+
+#endif
+
 namespace BloombergLP {
 namespace ntsu {
 
+namespace {
+
+#if defined(BSLS_PLATFORM_OS_LINUX)
+
+struct TimestampOtionSupport {
+    int option;
+    int versionMajor;
+    int versionMinor;
+    int versionPatch;
+};
+
+// clang-format off
+static TimestampOtionSupport s_timestampOptionSupport[] = {
+    { ntsu::TimestampUtil::e_SCM_TSTAMP_SND,                0,  0, 0 },
+    { ntsu::TimestampUtil::e_SCM_TSTAMP_SCHED,              0,  0, 0 },
+    { ntsu::TimestampUtil::e_SCM_TSTAMP_ACK,                0,  0, 0 },
+
+    { ntsu::TimestampUtil::e_SO_TIMESTAMPNS,                0,  0, 0 },
+    { ntsu::TimestampUtil::e_SO_TIMESTAMPING,               0,  0, 0 },
+    { ntsu::TimestampUtil::e_SCM_TIMESTAMPNS,               0,  0, 0 },
+    { ntsu::TimestampUtil::e_SCM_TIMESTAMPING,              0,  0, 0 },
+
+    { ntsu::TimestampUtil::e_SOF_TIMESTAMPING_TX_HARDWARE,  0,  0, 0 },
+    { ntsu::TimestampUtil::e_SOF_TIMESTAMPING_TX_SOFTWARE,  0,  0, 0 },
+    { ntsu::TimestampUtil::e_SOF_TIMESTAMPING_TX_SCHED,     3, 17, 0 },
```

Comment:
I thought you wanted to make a cut at TX timestamping usage being  supported only starting from 4.18. Then it's not clear why you want to have kernel 3.17 mentioned here.

Reply:
I'm not sure we need this facility; currently its preprocessed out. The idea was to detect the run-time kernel version and clear out the bits for unsupported flags. My testing shows that this isn't needed. I'm a bit paranoid about what we may find in the future, though, so I'd like to leave this system in even though it is preprocessed out of the code. In the meantime, I've stated the minimum necessary kernel version for all timestamping features is 4.18 (RHEL 8). 

If we find this system is necessary in the future, we'll go an exactly specify when each of the these flags were introduced. If we don't find this system necessary after some time has passed let's just remove it.